src/backend/distributed/connection/remote_commands.c

```diff
@@ -550,6 +541,12 @@ GetRemoteCommandResult(MultiConnection *connection, bool raiseInterrupts)
 
 	if (!FinishConnectionIO(connection, raiseInterrupts))
 	{
+		/* some error(s) happened while doing the I/O, signal the callers */
+		if (PQstatus(pgConn) == CONNECTION_BAD)
+		{
+			return PQmakeEmptyPGresult(pgConn, PGRES_FATAL_ERROR);
```

Comment:
Could we try setting the error message from `PQerrorMessage(pgConn)` here?

Reply:
`PQmakeEmptyPGresult ` seems to do that, so I skipped doing that